packages/eas-cli/src/credentials/android/actions/SetUpGoogleServiceAccountKeyForFcmV1.ts

```diff
@@ -0,0 +1,87 @@
+import nullthrows from 'nullthrows';
+
+import { AssignGoogleServiceAccountKeyForFcmV1 } from './AssignGoogleServiceAccountKeyForFcmV1';
+import { CreateGoogleServiceAccountKeyForFcmV1 } from './CreateGoogleServiceAccountKeyForFcmV1';
+import { UseExistingGoogleServiceAccountKey } from './UseExistingGoogleServiceAccountKey';
+import {
+  CommonAndroidAppCredentialsFragment,
+  GoogleServiceAccountKeyFragment,
+} from '../../../graphql/generated';
+import Log from '../../../log';
+import { promptAsync } from '../../../prompts';
+import { CredentialsContext } from '../../context';
+import { MissingCredentialsNonInteractiveError } from '../../errors';
+import { AppLookupParams } from '../api/GraphqlClient';
+
+export class SetUpGoogleServiceAccountKeyForFcmV1 {
```

Comment:
Let's refactor this in a similar way to the AssignGSAK class. We'll have a single `SetUpGoogleServiceAccountKey` that takes a parameter as input to determine whether we will set up the GSAK for submissions or fcmv1

Reply:
Similar reasoning to above, it seems premature/not super necessary to abstract this given there's only two use cases for it. There's definitely functionality overlap but it seems like a wash given there's only 2 use cases.